samples/kvssink_intermittent_sample.cpp

```diff
@@ -0,0 +1,293 @@
+        // We don't want to flush until the EOS is done to ensure all frames buffered in the pipeline have been processed.
+        cv.wait(lck);
+
+        // Flushing to remove EOS status.
+        GstEvent* flush_start = gst_event_new_flush_start();
+        gst_element_send_event(pipeline, flush_start);
+
+        // Using cv.wait_for to break sleep early upon signal interrupt. Checking for termination again before waiting.
+        if (terminated || cv.wait_for(lck, std::chrono::seconds(KVS_INTERMITTENT_PAUSED_INTERVAL_SECONDS)) != std::cv_status::timeout) {
+            break;
+        }
+
+        LOG_INFO("[KVS sample] Starting stream to KVS for " << KVS_INTERMITTENT_PLAYING_INTERVAL_SECONDS << " seconds");
+        GstEvent* flush_stop = gst_event_new_flush_stop(true);
+        gst_element_send_event(pipeline, flush_stop);
+    }
+    LOG_DEBUG("[KVS sample] Exited stopStartLoop");
+}
+
+int main (int argc, char *argv[])
+{
+    signal(SIGINT, sigint_handler);
+
+    CustomData customData;
+    GstElement *pipeline, *source, *clock_overlay, *video_convert, *source_filter, *encoder, *sink_filter, *kvssink;
+    GstCaps *source_caps, *sink_caps;
+    GstBus *bus;
+    guint bus_watch_id;
+    StreamSource source_type;
+    char stream_name[MAX_STREAM_NAME_LEN + 1];
+
+    gst_init (&argc, &argv);
+
+
+    /* Parse input arguments */
+
+    // Check for invalid argument count.
+    if (argc > 3 || argc < 2) {
+        LOG_ERROR("[KVS sample] Invalid argument count");
+        LOG_INFO("[KVS sample] Usage: " << argv[0] << " <streamName (required)> <testsrc or devicesrc>");
+        return -1;
+    }
+
+    // Get stream name.
+    STRNCPY(stream_name, argv[1], MAX_STREAM_NAME_LEN);
+    stream_name[MAX_STREAM_NAME_LEN] = '\0';
+
+    // Get source type.
+    if(argc > 2) {
+        if(0 == STRCMPI(argv[2], "testsrc")) {
+            LOG_INFO("[KVS sample] Using test source (videotestsrc)");
+            source_type = TEST_SOURCE;
+        } else if (0 == STRCMPI(argv[2], "devicesrc")) {
+            LOG_INFO("[KVS sample] Using device source (autovideosrc)");
+            source_type = DEVICE_SOURCE;
+        } else {
+            LOG_ERROR("[KVS sample] Invalid source type");
+            LOG_INFO("[KVS sample] Usage: " << argv[0] << " <streamName (required)> <testsrc or devicesrc>");
+            return -1;
+        }
+    } else {
+        LOG_ERROR("[KVS sample] No source specified, defualting to test source (videotestsrc)");
+        source_type = TEST_SOURCE;
+    }
+
+
+    /* Create GStreamer elements */
+
+    pipeline = gst_pipeline_new ("kvs-pipeline");
+
+    /* source */
+    if(source_type == TEST_SOURCE) {
+        source = gst_element_factory_make ("videotestsrc", "test-source");
+        g_object_set(G_OBJECT(source), "is-live", TRUE, NULL);
+    } else if (source_type == DEVICE_SOURCE) {
+        source = gst_element_factory_make ("autovideosrc", "device-source");
```

Comment:
Some places have a space between method and parenthesis and others not - please fix